File: 2026-08/src/day_29/prms_181890/Solution.java
package day_29.prms_181890;

import java.util.*;

class Solution {
    public String[] solution(String[] str_list) {
        int n = str_list.length;

        for (int i = 0; i < n; i++) {
            if (str_list[i].equals("l")) {
                return Arrays.copyOfRange(str_list, 0, i);
            }
            if (str_list[i].equals("r")) {
                return Arrays.copyOfRange(str_list, i + 1, n);
            }
        }

        return new String[0];
    }
}

File: 2026-08/src/day_29/prms_181890/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

vector<string> solution(vector<string> str_list) {
    int n = str_list.size();

    for (int i = 0; i < n; i++) {
        if (str_list[i] == "l") {
            return vector<string>(str_list.begin(), str_list.begin() + i);
        }
        if (str_list[i] == "r") {
            return vector<string>(str_list.begin() + i + 1, str_list.end());
        }
    }

    return {};
}
